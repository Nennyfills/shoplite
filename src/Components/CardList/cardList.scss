.card-list {
  background-color: #e3e6e3;
  .list-1 {
    display: flex;
    flex-wrap: wrap;
    max-width: 1300px;
    margin: auto;
  }
}
