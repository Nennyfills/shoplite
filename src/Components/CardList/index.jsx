import React from 'react';
import PropTypes from 'prop-types';

import Card from 'Components/Card';
import CardLoader from 'Components/CardLoader';

import './cardList.scss';

const CardList = ({ products, loading, getTotal, addAction }) => {
  return (
    <div className="card-list" data-test="card-list">
      <div className="list-1">
        {loading
          ? [1, 2, 3].map((_, index) => (
              <div key={index}>
                <CardLoader />
              </div>
            ))
          : products?.map((item) => (
              <div key={item.id}>
                <Card
                  src={item.image_url}
                  title={item.title}
                  currency={'$'}
                  price={item.price}
                  btnName={'Add to Cart'}
                  btnOnClick={() => {
                    addAction(item);
                    getTotal();
                  }}
                />
              </div>
            ))}
      </div>
    </div>
  );
};

CardList.defaultProps = {
  loading: true,
  products: [],
};

CardList.propTypes = {
  getTotal: PropTypes.func.isRequired,
  addAction: PropTypes.func.isRequired,
  products: PropTypes.array,
  loading: PropTypes.bool,
};

export default CardList;
