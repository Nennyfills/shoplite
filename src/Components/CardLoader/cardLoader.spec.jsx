import React from 'react';
import { shallow } from 'enzyme';

import CardList from '.';

const setUp = (props = {}) => {
  const component = shallow(<CardList {...props} />);
  return component;
};

describe('Card Loader', () => {
  let component;
  beforeEach(() => {
    component = setUp({});
  });
  it('Should render without errors', () => {
    expect(component).toMatchSnapshot();
  });
  it('Should render cardList div', () => {
    const wrapper = component.find(`[data-test='card-loader']`);
    expect(wrapper.length).toBe(1);
  });
});
