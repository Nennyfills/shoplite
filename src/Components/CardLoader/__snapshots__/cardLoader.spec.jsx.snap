// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Loader Should render without errors 1`] = `ShallowWrapper {}`;
